src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,719 @@
+            int probeBitrateBps =
+                    Math.min(bestProbe.getSendBitrateBps(), bestProbe
+                            .getRecvBitrateBps());
+            // Make sure that a probe sent on a lower bitrate
+            // than our estimate can't
+            // reduce the estimate.
+            if (isBitrateImproving(probeBitrateBps))
+            {
+                logger.warn("Probe successful, sent at "
+                        + bestProbe.getSendBitrateBps() +
+                        " bps, received at "
+                        + bestProbe.getRecvBitrateBps()
+                        + " bps. Mean send delta: " + bestProbe.sendMeanMs
+                        + " ms, mean recv delta: " + bestProbe.recvMeanMs
+                        + " ms, num probes: " + bestProbe.count);
+                remoteRate.setEstimate(probeBitrateBps, nowMs);
+                return ProbeResult.kBitrateUpdated;
+            }
+            // Not probing and received non-probe packet,
+            // or finished with current set  of probes.
+            if (clusters.size() >= kExpectedNumberOfProbes)
+                probes_.clear();
+            return ProbeResult.kNoUpdate;
+        }
+    }
+
+    boolean isBitrateImproving(int newBitrateBps)
+    {
+        synchronized (critSect)
+        {
+            boolean initialProbe = !remoteRate.isValidEstimate()
+                    && newBitrateBps > 0;
+            boolean bitrateAboveEstimate =
+                    remoteRate.isValidEstimate() &&
+                            newBitrateBps > (int) (remoteRate
+                                    .getLatestEstimate());
+            return initialProbe || bitrateAboveEstimate;
+        }
+    }
+
+    /**
+     * Reverse-transforms a specific packet.
+     *
+     * @param packet the transformed packet to be restored.
+     * @return the restored packet.
+     */
+    @Override
+    public RawPacket reverseTransform(
+            RawPacket packet)
+    {
+        if (absoluteSendTimeEngine
+                .hasAbsoluteSendTimeExtension(packet) == null) {
+            logger.warn( "RemoteBitrateEstimatorAbsSendTimeImpl: Incoming" +
+                    " packet is missing absolute send time extension!");
+        return null;
+    }
+        logger.info("Using RemoteBitrateEstimatorAbsSendTime: Instantiating.");
+        incomingPacketInfo(System.currentTimeMillis(), absoluteSendTimeEngine
+                        .getAbsSendTime(packet),
+            packet.getPayloadLength(), packet.getSSRC());
+        return packet;
+    }
+
+    void incomingPacketInfo(
+        long arrivalTimeMs,
+        long sendTime24bits,
+        long payloadSize,
+        long ssrc) {
+        /**
+         * @ToDo 1L << 24 should be 1L << 23 since we want to check if
+         * sendTime24bits is less than 24 bits
+         */
+        if (sendTime24bits < (1L << 24)){
+            logger.warn("Send Time not valid");
```

Comment:
This check doesn't seem right. Valid values should be [0, 2^24-1], so the check should be something like if (sendTime24bits < 0 || sendTime24bits >= (1 << 24))